treq/auth.py

```diff
@@ -26,6 +182,112 @@ def request(self, method, uri, headers=None, bodyProducer=None):
             method, uri, headers=headers, bodyProducer=bodyProducer)
 
 
+class _RequestDigestAuthenticationAgent(object):
+
+    digest_auth_cache = {}
```

Comment:
This also shouldn't be public. Is this really a class variable? It looks like it's only accessed via the instance so it should probably be initialized in in `__init__`.